File: tests/fixtures/string_ops.clj
; Test: String operations

class StringOpsTest {
    sfn m@(main):void () {
        def text "Hello World"
        
        ; String length
        def len (strlen text)
        print ("Length: " + len)
        
        ; Substring
        def sub (substring text 0 5)
        print sub
        
        ; String concatenation
        def combined (text + "!")
        print combined
        
        ; Index of
        def idx (indexOf text "World")
        print ("Index: " + idx)
        
        print "Done"
    }
}

File: tests/.output-kotlin/string_ops.kt

class StringOpsTest {
  companion object {
    
  }
}

fun main(args : Array<String>) {
  val text : String  = "Hello World";
  val __len : Int  = text.length;
  println( "Length: " + __len.toString() )
  val sub : String  = text.substring(0, 5 );
  println( sub )
  val combined : String  = text + "!";
  println( combined )
  val idx :   = indexOftext"World";
  print+"Index: "idx
  println( "Done" )
}
